modelon/impact/client/entities/workspace.py

```diff
@@ -1126,3 +1210,53 @@ def from_conversion_operation(
     ) -> Workspace:
         assert isinstance(operation, WorkspaceConversionOperation)
         return cls(**kwargs, service=operation._sal)
+
+    @property
+    def received_from(self) -> ReceivedFromWorkspace:
+        """Returns the ReceivedFromWorkspace class."""
+        return ReceivedFromWorkspace(self)
```

Comment:
Should be returning None if there is not received_from workspace.

Reply:
If we are raising an exception in the constructor, would this be required? Instead if we let the user get an exception that would be more informative for them right? For example if I have code like workspace.recieved_from.get_workspace() I would get an exception like "None" type has no attribute "get_workspace" which I feel is less intuitive to non-programers.

Reply:
agreed